src/server/mod.rs

```diff
@@ -568,14 +576,18 @@ impl<C: XConnection> ServerState<C> {
 
         if win.attrs.size_hints.is_none() || *win.attrs.size_hints.as_ref().unwrap() != hints {
             debug!("setting {window:?} hints {hints:?}");
-            if let Some(surface) = win.surface_key {
-                let surface: &SurfaceData = self.objects[surface].as_ref();
-                if let Some(SurfaceRole::Toplevel(Some(data))) = &surface.role {
-                    if let Some(min_size) = &hints.min_size {
-                        data.toplevel.set_min_size(min_size.width, min_size.height);
-                    }
-                    if let Some(max_size) = &hints.max_size {
-                        data.toplevel.set_max_size(max_size.width, max_size.height);
+            if let Some(key) = win.surface_key {
+                if let Some(object) = self.objects.get(key) {
+                    let surface: &SurfaceData = object.as_ref();
+                    if let Some(SurfaceRole::Toplevel(Some(data))) = &surface.role {
+                        if let Some(min_size) = &hints.min_size {
+                            data.toplevel.set_min_size(min_size.width, min_size.height);
+                        }
+                        if let Some(max_size) = &hints.max_size {
+                            data.toplevel.set_max_size(max_size.width, max_size.height);
+                        }
+                    } else {
+                        warn!("Could not set size hint on {window:?}: stale surface")
```

Comment:
Isn't this else at the wrong level? I think it'll even warn for size hints on popups.

Reply:
fixed